jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ConfigFilesFacade.kt

```diff
@@ -164,17 +102,76 @@ object DefaultConfigFileWriter : ConfigFileWriter {
         # [sso-session my-sso-portal-us-east-1]
         # sso_region = us-east-1
         # sso_start_url = https://my-sso-portal.awsapps.com/start
-        """.trimIndent()
+            """.trimIndent()
+    }
 
-    override fun createFile(file: File) {
-        val path = file.toPath()
-        path.tryDirOp(LOG) { createParentDirectories() }
+    override fun readAllProfiles(): Map<String, Profile> = ProfileFile.aggregator()
+        .applyMutation {
+            if (credentialsPath.exists()) {
+                it.addFile(
+                    ProfileFile.builder()
+                        .content(credentialsPath)
+                        .type(ProfileFile.Type.CREDENTIALS)
+                        .build()
+                )
+            }
+        }
+        .applyMutation {
+            if (configPath.exists()) {
+                it.addFile(
+                    ProfileFile.builder()
+                        .content(configPath)
+                        .type(ProfileFile.Type.CONFIGURATION)
+                        .build()
+                )
+            }
+        }
+        .build()
+        .profiles()
 
-        path.tryFileOp(LOG) {
+    override fun createConfigFile() {
+        configPath.tryDirOp(LOG) { createParentDirectories() }
+
+        configPath.tryFileOp(LOG) {
             touch(restrictToOwner = true)
             writeText(TEMPLATE)
         }
     }
 
-    private val LOG = getLogger<DefaultConfigFileWriter>()
+    override fun appendProfileToConfig(profile: Profile) =
+        appendSection(configPath, "profile", profile)
+
+    override fun appendProfileToCredentials(profile: Profile) =
+        appendSection(credentialsPath, "profile", profile)
+
+    override fun appendSectionToConfig(sectionName: String, profile: Profile) =
+        appendSection(configPath, sectionName, profile)
+
+    private fun appendSection(path: Path, sectionName: String, profile: Profile) {
+        val isConfigFile = path.fileName.toString() != "credentials"
+        if (sectionName == "sso-session" && !isConfigFile) {
+            error("sso-session is only allowed in 'config'")
+        }
+
+        // behavior is different between config/credentials
+        val sectionTitle = if (!isConfigFile || profile.name().trim() == "default") {
```

Comment:
nit: For reference could we add a comment with an example of the structure of this section